tools/clang/tools/dxcompiler/dxcpdbutils.cpp

```diff
@@ -752,6 +749,14 @@ struct DxcPdbUtils : public IDxcPdbUtils, public IDxcPixDxilDebugInfoFactory
     return m_pDebugProgramBlob != nullptr;
   }
 
+  virtual void STDMETHODCALLTYPE SetEntryPointToDefaultIfEmpty() {
+    // Entry point might have been omitted. Set it to main by default.
+    // TODO: Check to see that this DxilContainer is not a library before setting the entry point.
+    if (m_EntryPoint.empty()) {
```

Comment:
It's possible that the input is just a non-debug DXIL and have no args at all, in which case we don't want to populate m_EntryPoint. Should also check if m_ArgPairs is not empty.